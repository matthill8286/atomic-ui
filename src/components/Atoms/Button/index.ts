export * from './Button'
export * from './Button.interface'
export * from './IconButton'
