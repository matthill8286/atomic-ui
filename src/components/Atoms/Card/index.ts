export * from './CardV2'
export * from './Card.interface'
export { StyledCard } from './Card.styled'
