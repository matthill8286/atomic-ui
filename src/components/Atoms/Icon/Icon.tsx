import { FlexAlignItemsOptions } from '@/components/Helper/FlexBox'
import { padding } from '@/styles/sc-shared-functions'
import { css, styled } from '@/styles/styled'
import { PaddingProps, Size, ThemeColors } from '@/types'

export type RotationValues = 0 | 45 | 90 | 180 | 270 | -45 | -90 | -180 | -270

export interface IconProps {
  rotate?: RotationValues
  color?: ThemeColors
  withBorder?: boolean
  padding?: PaddingProps['padding']

  width?: number | keyof Omit<Size, 'xxxs'>
  height?: number | keyof Omit<Size, 'xxxs'>
  animate?: boolean
  display?: string
  alignSelf?: FlexAlignItemsOptions
}

const DefaultSize = '25px'

export const Icon = styled.div<IconProps>(
  ({
    theme,
    color,
    width,
    height,
    rotate = 0,
    animate = false,
    display = 'flex',
    alignSelf,
    withBorder,
  }) => css`
    ${display && `display: ${display};`};
    ${padding};
    ${alignSelf && `align-self: ${alignSelf};`};
    ${withBorder &&
      `border: 2px solid ${theme.color.grey2};
        border-radius: 50%;
        padding: ${theme.spacing.base.xs};`}
    svg {
      width: ${width ? getSizeInPx(width, theme.spacing.base) : DefaultSize};
      height: ${height ? getSizeInPx(height, theme.spacing.base) : DefaultSize};
      transform: rotate(${rotate}deg);
      fill: ${color ? theme.color[color] : 'currentColor'};
      ${animate &&
        css`
          transition: transform 0.3s ease-in-out;
        `};
    }
    svg g,
    svg p {
      fill: ${color ? theme.color[color] : 'currentColor'};
    }
  `
)

const getSizeInPx = (
  size: number | keyof Omit<Size, 'xxxs'>,
  themeSizes: Omit<Size, 'xxxs'>
): string => {
  if (typeof size === 'number') {
    return `${size}px`
  } else {
    return themeSizes[size] || `${DefaultSize}px`
  }
}
