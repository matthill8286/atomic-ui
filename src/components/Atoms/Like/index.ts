export * from './Like'
export * from './Like.interface'
