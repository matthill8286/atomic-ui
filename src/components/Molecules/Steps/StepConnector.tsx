import React, { FC } from 'react'
import { css, styled } from '@/styles/styled'
import {
  StepConnectorLineProps,
  StepConnectorPositions,
  StepConnectorProps,
} from './Steps.interface'

const getStepConnectorPositions = ({
  theme,
  variant,
  elementPosition,
  rightMargin,
  leftMargin,
}): StepConnectorPositions => {
  const positions: StepConnectorPositions = {
    uniform: {
      first: css`
        left: 50%;
        right: 0;
      `,
      second: css`
        left: -50%;
        right: 50%;
      `,
      last: css`
        left: -50%;
        right: 50%;
      `,
      default: css`
        left: -50%;
        right: 50%;
      `,
    },
    prominent: {
      first: css`
        left: 10px;
        right: calc(100% - ${theme.spacing.base.md});
      `,
      second: css`
        left: calc(-100% + ${theme.spacing.base.md});
        right: 50%;
      `,
      last: css`
        left: -50%;
        right: 10px;
      `,
      default: css`
        left: -50%;
        right: 50%;
      `,
    },
    simple: {
      first: css`
        display: none;
      `,
      second: css`
        left: calc(-100% + ${42 + leftMargin}px);
        right: calc(50% + ${28 + rightMargin}px);
      `,
      last: css`
        left: calc(-50% + ${28 + leftMargin}px);
        right: calc(0% + ${42 + rightMargin}px);
      `,
      default: css`
        left: calc(-50% + ${28 + leftMargin}px);
        right: calc(50% + ${28 + rightMargin}px);
      `,
    },
  }

  return positions[variant][elementPosition]
}

const BottomPositions = {
  uniform: {
    completed: '16px',
    default: '8px',
  },
  prominent: {
    completed: '16px',
    default: '8px',
  },
  simple: {
    completed: '5px',
    default: '5px',
  },
}

export const StepConnectorRoot = styled.div<StepConnectorProps>(
  ({
    theme,
    elementPosition = 'default',
    active,
    completed,
    prevActive,
    variant = 'prominent',
  }) => {
    const leftMargin = active ? 4 : 0
    const rightMargin = prevActive ? 4 : 0

    const currentPosition = getStepConnectorPositions({
      theme,
      variant,
      elementPosition,
      leftMargin,
      rightMargin,
    })

    const bottomPosition = completed
      ? BottomPositions[variant]['completed']
      : BottomPositions[variant]['default']

    return css`
      position: absolute;
      bottom: ${bottomPosition};
      ${currentPosition};
    `
  }
)

export const StepConnectorLine = styled.span<StepConnectorLineProps>`
  border-color: ${({ theme, color = 'primary', active, completed, elementPosition = 'default' }) =>
    completed || active || elementPosition === 'first' ? theme.color[color] : theme.color.grey2};
  border-style: solid;
  border-top-width: ${({ borderWidth }) => borderWidth};
  border-bottom-width: ${({ borderWidth }) => borderWidth};
  border-radius: ${({ theme }) => theme.dimension.borderRadius1};
  display: block;
`

export const StepConnector: FC<StepConnectorProps> = props => (
  <StepConnectorRoot {...props}>
    <StepConnectorLine {...props} borderWidth={props.variant === 'simple' ? '2px' : '4px'} />
  </StepConnectorRoot>
)

StepConnector.displayName = 'StepConnector'
