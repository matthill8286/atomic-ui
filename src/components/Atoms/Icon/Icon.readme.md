## How to use?
Import the `Icon` component and also import the SVG you want to use directly from the `@excelwithbusiness/webmobile-svg-library` module and nest it. The `Icon` component itself has the Prop `rotate` to allow a rotation of a SVG.

```javascript
import { Icon } from '@/components/Atoms/Icon'
import { IconArrow } from '@excelwithbusiness/webmobile-svg-library'

<Icon>
  <IconArrow />
</Icon>

<Icon rotate={90}> 
  <IconArrow />
</Icon>
```

  &nbsp;

## List of existing Icons

```js

IconAdd

IconArticle

IconBackUi

IconBookmark

IconBookmarkActive

IconCheckmarkCircle

IconCheckmarkCircleOutlined

IconCheckmarkSquare

IconCheckmarkSquareOutlined

IconClear

IconClearCircle

IconClearCircleOutlined

IconClearSquare

IconClearSquareOutlined

IconCross

IconDone

IconFastForward

IconFastRewind

IconFilter

IconForward

IconHeart

IconHeartOutlined

IconHelp

IconHelpOutlined

IconInfo

IconInfoOutined

IconMenuIcon

IconNext

IconPause

IconPlay

IconPlayOutlined

IconSearch

IconSettings

IconSettingsOutlined

IconSkip

IconSkipNext

IconSkipPrevious

IconVolumeDown

IconVolumeLow

IconVolumeMinus

IconVolumeMute

IconVolumeOff

IconVolumePlus

IconVolumeUp

```
