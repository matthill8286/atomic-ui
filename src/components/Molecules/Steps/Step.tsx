import React, { FC } from 'react'
import { Typo } from '@/components/Atoms/Typography'
import { media } from '@/styles/media'
import { styled } from '@/styles/styled'
import { ThemeColors } from '@/types/theme'
import { StepIcon } from './StepIcon'
import {
  ElementPosition,
  StepIconProps,
  StepPrivateProps,
  StepPublicProps,
} from './Steps.interface'

const StepRoot = styled.div<{ pointer: boolean }>`
  cursor: ${({ pointer }) => (pointer ? 'pointer' : 'default')};
  flex: 1;
  position: relative;
  padding: 0 ${({ theme }) => theme.defaultSpacing}px;
`

const StepLabelRoot = styled.span<{ elementPosition: ElementPosition }>`
  flex-direction: column;
  display: flex;
  align-items: center;

  ${({ elementPosition }) => elementPosition === 'first' && 'align-items: flex-start;'}
  ${({ elementPosition }) => elementPosition === 'last' && 'align-items: flex-end;'}
`

const StepDescription = styled(Typo)`
  ${media.maxMd} {
    display: none;
  }
`
export const Step: FC<StepPublicProps & StepPrivateProps> = props => {
  const {
    color = 'primary',
    elementPosition = 'default',
    active = false,
    completed = false,
    stepIndex = 0,
    title = '',
    description = '',
    icon,
    connector,
    onClick,
    variant = 'prominent',
    ...other
  } = props

  const stepIconProps: StepIconProps = {
    active,
    completed,
    color,
    variant,
    customIcon: icon,
  }

  const textColor: ThemeColors = active || completed ? color : 'grey2'

  return (
    <StepRoot {...other} pointer={!!onClick} onClick={() => onClick && onClick(stepIndex)}>
      {connector}
      {variant != 'uniform' ? (
        <StepLabelRoot elementPosition={elementPosition}>
          {title && (
            <Typo
              tag="span"
              color={textColor}
              weight={active || completed ? 'semibold' : 'regular'}>
              {title}
            </Typo>
          )}
          {description && (
            <StepDescription tag="span" color={textColor}>
              {description}
            </StepDescription>
          )}
          <StepIcon {...stepIconProps} />
        </StepLabelRoot>
      ) : (
        <StepIcon {...stepIconProps} />
      )}
    </StepRoot>
  )
}

Step.displayName = 'Step'
