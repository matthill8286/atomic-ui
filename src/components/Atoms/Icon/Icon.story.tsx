import { storiesOf } from '@storybook/react'
import React from 'react'
import { FlexBox } from '@/components/Helper/FlexBox'
import {
  OtherArrowRightCircle,
  OtherBookmark,
  OtherClose,
  OtherCompleteChecked,
  OtherDone,
  OtherUseful,
  OtherArticle,
  OtherFacebook,
} from '@matthill8286/atomic-icon-library'
import { Icon } from './Icon'
import { Spacer } from '@/components/Atoms/Spacer'

import Readme from './Icon.readme.md'
import { IconButton } from '@/components/Atoms/Button'

storiesOf('Design System/Atoms/Icon', module)
  .add(
    'Icons',
    () => {
      return (
        <>
          <Icon color={'primary'}>
            <OtherArticle />
          </Icon>
          <Spacer size={'xl'} />
          <Icon rotate={90}>
            <OtherCompleteChecked />
          </Icon>
          <Spacer size={'xl'} />
          <Icon height="md" width="md">
            <OtherBookmark />
          </Icon>
          <Spacer size={'xl'} />
          <Icon height="sm" width="sm">
            <OtherClose />
          </Icon>
          <Spacer size={'xl'} />
          <Icon height="lg" width="lg" color={'secondary'}>
            <OtherUseful height="lg" width="lg" />
          </Icon>
          <Spacer size={'xl'} />
          <Icon height={'md'} width={'md'} color={'primary'}>
            <OtherArrowRightCircle />
          </Icon>
        </>
      )
    },
    {
      info: Readme,
    }
  )
  .add(
    'Rotated icon',
    () => {
      return (
        <Icon height={'md'} width={'md'} rotate={90}>
          <OtherArrowRightCircle fill={'primary'} />
        </Icon>
      )
    },
    {
      info: Readme,
    }
  )
  .add('With border', () => {
    return (
      <FlexBox>
        <Icon withBorder width={25} height={25} color="primary">
          <OtherFacebook />
        </Icon>
      </FlexBox>
    )
  })
  .add('Large primary icon button', () => {
    return (
      <FlexBox>
        <IconButton round width={65} height={65} actionType="primary">
          <OtherDone fill={'white'} />
        </IconButton>
      </FlexBox>
    )
  })
