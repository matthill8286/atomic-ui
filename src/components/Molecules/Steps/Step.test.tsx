import React, { FC } from 'react'
import { Icon } from '@/components/Atoms/Icon'
import { styled } from '@/styles/styled'
import { IconArticle } from '@excelwithbusiness/webmobile-svg-library'
import { mountWithTheme, renderWithTheme } from '@/testRenderer'
import { ThemeColors } from '@/types/theme'
import { Step } from './Step'
import { StepConnectorRoot } from './StepConnector'
import {
  StepConnectorLineProps,
  StepConnectorProps,
  StepPrivateProps,
  StepPublicProps,
} from './Steps.interface'

const props: StepPublicProps & StepPrivateProps = {
  title: 'Sign In',
  description: 'Test description',
  color: 'primary',
  active: false,
  completed: false,
  elementPosition: 'first',
  stepIndex: 0,
  variant: 'prominent',
}

describe('Step', () => {
  it('should render', () => {
    const tree = renderWithTheme(<Step {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('should render with custom icon', () => {
    interface CustomIconProps {
      active?: boolean
      completed?: boolean
      color?: ThemeColors
    }

    const CustomIcon: FC<CustomIconProps> = ({
      active = false,
      completed = false,
      color = 'primary',
      ...props
    }) => (
      <Icon color={active ? 'white' : completed ? color : 'grey2'} {...props}>
        <IconArticle />
      </Icon>
    )

    const withIconProps = {
      ...props,
      icon: <CustomIcon />,
    }
    const tree = renderWithTheme(<Step {...withIconProps} />)
    expect(tree).toMatchSnapshot()
  })

  it('should render with custom connector', () => {
    const CustomConnectorLine = styled.span<StepConnectorLineProps>`
      border-color: ${({
        theme,
        color = 'primary',
        active,
        completed,
        elementPosition = 'default',
      }) =>
        completed || active || elementPosition === 'first'
          ? theme.color[color]
          : theme.color.black};
      border-style: solid;
      border-top-width: ${({ borderWidth }) => borderWidth};
      border-bottom-width: ${({ borderWidth }) => borderWidth};
      border-radius: 4px;
      display: block;
      position: 'relative';
      z-index: 'auto';
    `

    const CustomConnector: FC<StepConnectorProps> = props => (
      <StepConnectorRoot {...props}>
        <CustomConnectorLine {...props} borderWidth={props.variant === 'simple' ? '2px' : '4px'} />
      </StepConnectorRoot>
    )

    const withIconProps = {
      ...props,
      connector: <CustomConnector />,
    }
    const tree = renderWithTheme(<Step {...withIconProps} />)
    expect(tree).toMatchSnapshot()
  })

  it('should call onClick function', () => {
    const onClick = jest.fn()
    const three = mountWithTheme(<Step {...props} onClick={onClick} />)
    three.childAt(0).simulate('click')
    expect(onClick.mock.calls.length).toEqual(1)
  })
})
