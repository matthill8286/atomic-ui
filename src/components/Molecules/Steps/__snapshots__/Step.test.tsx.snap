// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step should render 1`] = `
.c2 {
  color: #E5E5E5;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c6 {
  color: #00C8DC;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c4 {
  margin-top: 8px;
  margin-bottom: 0;
  position: relative;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c5 {
  border: 4px solid #E5E5E5;
  border-radius: 50%;
  width: 24px;
  height: 24px;
  background-color: #E5E5E5;
  box-shadow: 0 0 0 #E5E5E5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 {
  cursor: default;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  padding: 0 8px;
}

.c1 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

@media (max-width:752px) {
  .c3 {
    display: none;
  }
}

<div
  className="c0"
  onClick={[Function]}
>
  <span
    className="c1"
  >
    <span
      className="c2"
      color="grey2"
    >
      Sign In
    </span>
    <span
      className="c2 c3"
      color="grey2"
    >
      Test description
    </span>
    <div
      className="c4"
    >
      <div
        className="c5"
        color="primary"
      >
        <span
          className="c6"
          color="primary"
        />
      </div>
    </div>
  </span>
</div>
`;

exports[`Step should render with custom connector 1`] = `
.c4 {
  color: #E5E5E5;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c8 {
  color: #00C8DC;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c6 {
  margin-top: 8px;
  margin-bottom: 0;
  position: relative;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c7 {
  border: 4px solid #E5E5E5;
  border-radius: 50%;
  width: 24px;
  height: 24px;
  background-color: #E5E5E5;
  box-shadow: 0 0 0 #E5E5E5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 {
  cursor: default;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  padding: 0 8px;
}

.c3 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  position: absolute;
  bottom: 8px;
  left: -50%;
  right: 50%;
}

.c2 {
  border-color: #000000;
  border-style: solid;
  border-top-width: 4px;
  border-bottom-width: 4px;
  border-radius: 4px;
  display: block;
  position: 'relative';
  z-index: 'auto';
}

@media (max-width:752px) {
  .c5 {
    display: none;
  }
}

<div
  className="c0"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <span
      className="c2"
    />
  </div>
  <span
    className="c3"
  >
    <span
      className="c4"
      color="grey2"
    >
      Sign In
    </span>
    <span
      className="c4 c5"
      color="grey2"
    >
      Test description
    </span>
    <div
      className="c6"
    >
      <div
        className="c7"
        color="primary"
      >
        <span
          className="c8"
          color="primary"
        />
      </div>
    </div>
  </span>
</div>
`;

exports[`Step should render with custom icon 1`] = `
.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 svg {
  width: 25px;
  height: 25px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: #E5E5E5;
}

.c6 svg g,
.c6 svg p {
  fill: #E5E5E5;
}

.c2 {
  color: #E5E5E5;
  font-family: "Oswald","Raleway",Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c4 {
  margin-top: 8px;
  margin-bottom: 0;
  position: relative;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c5 {
  border: 4px solid #E5E5E5;
  border-radius: 50%;
  width: 24px;
  height: 24px;
  background-color: #E5E5E5;
  box-shadow: 0 0 0 #E5E5E5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 {
  cursor: default;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  position: relative;
  padding: 0 8px;
}

.c1 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

@media (max-width:752px) {
  .c3 {
    display: none;
  }
}

<div
  className="c0"
  onClick={[Function]}
>
  <span
    className="c1"
  >
    <span
      className="c2"
      color="grey2"
    >
      Sign In
    </span>
    <span
      className="c2 c3"
      color="grey2"
    >
      Test description
    </span>
    <div
      className="c4"
    >
      <div
        className="c5"
        color="primary"
      >
        <div
          className="c6"
          color="grey2"
        >
          <svg
            height={16}
            width={16}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.944 0a.74.74 0 01.106.007L9.944 0a.733.733 0 01.442.148l.07.06 4.332 4.267c.025.024.048.05.069.078l-.069-.078a.714.714 0 01.128.17l.011.022a.67.67 0 01.033.077l.007.023a.647.647 0 01.033.21v8.534c0 1.178-.97 2.133-2.167 2.133H4.167C2.97 15.644 2 14.69 2 13.511V2.133C2 .955 2.97 0 4.167 0zm-.722 1.422H4.167a.717.717 0 00-.723.711v11.378c0 .393.324.711.723.711h8.666a.717.717 0 00.723-.71l-.001-7.824h-3.61a.719.719 0 01-.716-.614l-.007-.096V1.422zm2.167 9.245c.399 0 .722.318.722.71a.717.717 0 01-.722.712H5.61a.717.717 0 01-.722-.711c0-.393.323-.711.722-.711zm0-2.845c.399 0 .722.319.722.711a.717.717 0 01-.722.711H5.61a.717.717 0 01-.722-.71c0-.393.323-.712.722-.712zM7.056 4.978c.398 0 .722.318.722.71a.717.717 0 01-.722.712H5.61a.717.717 0 01-.722-.711c0-.393.323-.711.722-.711zm3.61-2.551v1.84l1.867-.001-1.867-1.839z"
              fill="#666"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </span>
</div>
`;
